### PR TITLE
Tidy up `CallInst`.

### PR DESCRIPTION
**Pull request by ltratt**
This is a follow-up to https://github.com/ykjit/yk/pull/1130, further tidying up `CallInst`. It's mostly fairly mechanical.

### CODE REVIEW

File: ykrt/src/compile/jitc_yk/jit_ir.rs [hunk 1]
```diff
@@ -238,16 +232,15 @@ impl Module {
...
-    /// If `ops` would overflow the index type.
-    fn push_extra_args(&mut self, ops: &[Operand]) -> Result<ExtraArgsIdx, CompilationError> {
-        assert!(ExtraArgsIdx::new(self.types.len()).is_ok());
-        let idx = self.extra_args.len();
-        self.extra_args.extend_from_slice(ops); // FIXME: this clones.
-        ExtraArgsIdx::new(idx)
+    /// If `args` would overflow the index type.
+    fn push_args(&mut self, args: &[Operand]) -> Result<ArgsIdx, CompilationError> {
+        let idx = self.args.len();
+        self.args.extend_from_slice(args); // FIXME: this clones.
```
Comment (vext01) [inline ykrt/src/compile/jitc_yk/jit_ir.rs]: I was staring at this, scratching my head thinking, "I don't see a `self.args`"... I must be losing the plot.

Checked out your branch, and indeed, `self.args` is only introduced in the next commit!

Comment (ltratt) [reply ykrt/src/compile/jitc_yk/jit_ir.rs]: Erp. I might have made a mistake when breaking this up. I can probably fix this in a rebase.

Comment (vext01) [reply ykrt/src/compile/jitc_yk/jit_ir.rs]: If it isn't too annoying, please go ahead. Might make it easier to review.

Comment (ltratt) [reply ykrt/src/compile/jitc_yk/jit_ir.rs]: OK, force pushed an update. Once I realised what (silly) mistake I'd made in splitting things up, it wasn't too bad!

Comment (vext01) [reply ykrt/src/compile/jitc_yk/jit_ir.rs]: Thanks!

**Comment by vext01** [comment]
This all makes sense now. merging.

**Pull request merged**